### PR TITLE
Expose the SemanticClassificationViewTagger for all Roslyn languages.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/12732

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by balajikris** [comment]
:+1:

File: src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs [hunk 1]
```diff
@@ -92,15 +91,14 @@ protected override Task ProduceTagsAsync(TaggerContext<IClassificationTag> conte
             Debug.Assert(context.SpansToTag.IsSingle());
 
             var spanToTag = context.SpansToTag.Single();
-            var document = spanToTag.Document;
 
-            if (document == null)
+            _classificationService = _classificationService ?? 
+                spanToTag.Document?.Project.LanguageServices.GetService<IEditorClassificationService>();
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs]: This is a bit subtle for my tastes - it isn't really clear that `Document` being null is an expected condition.

File: src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs [hunk 2]
```diff
@@ -29,8 +29,7 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Classification
     /// </summary>
     [Export(typeof(IViewTaggerProvider))]
     [TagType(typeof(IClassificationTag))]
-    [ContentType(ContentTypeNames.CSharpContentType)]
-    [ContentType(ContentTypeNames.VisualBasicContentType)]
+    [ContentType(ContentTypeNames.RoslynContentType)]
```
Comment (Pilchie) [inline src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs]: I noticed in @mgoertz-msft's PR #12948 there was a bunch of places Xaml was added to a set of `[ContentType()]` blocks.  Maybe we could apply this approach more widely?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs]: Agreed.  But i'd like to do them one at a time.

File: src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs]: I'll make less subtle :)

**Comment by CyrusNajmabadi** [comment]
It looks lke the other two cases are:

1. The service we use to fade out text.  I imagine every language would like this.
2. The service we use to offer 'organize imports'.  This would be valuable for F# and TS as well.

I'll work on doing these separately.

**Comment by Pilchie** [comment]
:+1:

File: src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs [hunk 3]
```diff
@@ -92,14 +91,20 @@ protected override Task ProduceTagsAsync(TaggerContext<IClassificationTag> conte
             Debug.Assert(context.SpansToTag.IsSingle());
 
             var spanToTag = context.SpansToTag.Single();
-            var document = spanToTag.Document;
 
-            if (document == null)
+            // Attempt to get a classification service which will actually produce the results.
+            // If we can't (because we have no Document, or because the language doesn't support
+            // this service), then bail out immediately.
+            if (_classificationService == null)
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs]: If we don't have a service, should we even have created a tagger in the first place? If the CreateTagger returns null to the editor, we won't even be participating.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationViewTaggerProvider.cs]: How would we know at the time?  The tagger is created on top of buffers/views, and may happen well before Roslyn has hooked up and we've gotten actual documents.

**Pull request merged**